src/components/structures/RoomView.js

```diff
@@ -375,6 +376,21 @@ module.exports = React.createClass({
         }
     },
 
+    onSearchContextFillRequest: function(backwards) {
+        if (!this.state.searchContext.getPaginateToken(backwards)) {
+            debuglog("no more context results");
+            return;
+        }
+
+        debuglog("requesting more search context; backwards:"+backwards);
+
+        return MatrixClientPeg.get().paginateEventContext(
+                this.state.searchContext, {backwards: backwards}
+        ).then(() => {
+            this.forceUpdate();
```

Comment:
It's a damn sight less unintuitive than the existing code for handling normal backfill, which does the same thing via a no-op setState:

```
 this.setState({
            room: MatrixClientPeg.get().getRoom(this.props.roomId)
        });
```

But I take your point. Will add a comment.